### ISSUE TITLE
Application closes when trying to disable Windows Module Installer

### ISSUE THREAD

**Issue by lukasIO** [WereDev/Wu10Man#4]
I just tried Wu10Man on ~ 20 machines and on half of them it was working properly, the other half didn't show a info box confirming the disabling of the windows update service, but seemed to disable it (at least the UI reflects it in that way). But when trying to also disable the Windows Module Installer on those machines the Program just exits wihtout any prompt (also no reopening as referred to in https://github.com/WereDev/Wu10Man/issues/1).
edit: actually it reopens on those machines.
I click "disable" -> application quickly closes and reopens... after ~3 attempts of that the application then just closes without reopening

**Comment by WereDev** [comment]
Is there any sort of error message or anything that displays?  There might be something in the Windows Event Viewer.

**Comment by WereDev** [comment]
@lukasIO I was able to replicate this though I obviously don't know if it's the same issue that you're having.  I just uploaded v1.1 that adds logging to the application so that if there is an exception being eaten, it should now end up in the logs.  Please try the updated version and if you still see the issue, include a copy of that log file here.

Thanks.

**Comment by lukasIO** [comment]
@WereDev thanks for looking at this... I used the new version and the log for the first error is
```
2018-06-22 19:10:19.0862 INFO Application started
2018-06-22 19:10:27.1156 ERROR Der Kontoname ist unzulässig oder nicht vorhanden, oder das Kennwort ist für den angegebenen Kontennamen ungültig 
    bei WereDev.Utils.Wu10Man.Editors.ServiceCredentialsEditor.SetWindowsServiceCreds(String serviceName, String username, String password) in C:\_Code\Wu10Man\Wu10Man\Editors\ServiceCredentialsEditor.cs:Zeile 38.
	   bei WereDev.Utils.Wu10Man.Editors.ServiceEditor.SetAccountAsLocalService() in C:\_Code\Wu10Man\Wu10Man\Editors\ServiceEditor.cs:Zeile 54.
	   bei WereDev.Utils.Wu10Man.Editors.ServiceEditor.DisableService() in C:\_Code\Wu10Man\Wu10Man\Editors\ServiceEditor.cs:Zeile 29.
	   bei WereDev.Utils.Wu10Man.UserControls.WindowsServicesControl.DisableService(String serviceName) in C:\_Code\Wu10Man\Wu10Man\UserControls\WindowsServicesControl.xaml.cs:Zeile 57.
	   bei WereDev.Utils.Wu10Man.UserControls.WindowsServicesControl.BtnEnableDisable_Click(Object sender, RoutedEventArgs e) in C:\_Code\Wu10Man\Wu10Man\UserControls\WindowsServicesControl.xaml.cs:Zeile 70.
```

there is a second error showing now, when trying to select "disable updates":
```
2018-06-22 19:11:32.1085 ERROR Der Objektverweis wurde nicht auf eine Objektinstanz festgelegt. 
    bei WereDev.Utils.Wu10Man.Editors.RegistryEditor.WriteRegistryValue(RegistryKey registryRoot, String registryKey, String registryName, String registryValue, RegistryValueKind registryValueKind) in C:\_Code\Wu10Man\Wu10Man\Editors\RegistryEditor.cs:Zeile 64.
	   bei WereDev.Utils.Wu10Man.Editors.RegistryEditor.WriteLocalMachineRegistryValue(String registryKey, String registryName, String registryValue, RegistryValueKind registryValueKind) in C:\_Code\Wu10Man\Wu10Man\Editors\RegistryEditor.cs:Zeile 26.
	   bei WereDev.Utils.Wu10Man.UserControls.GroupPolicyControl.btnSetGroupPolicy_Click(Object sender, RoutedEventArgs e) in C:\_Code\Wu10Man\Wu10Man\UserControls\GroupPolicyControl.xaml.cs:Zeile 112.
```

**Comment by WereDev** [comment]
@lukasIO The first one looks like an error trying to find the LocalService user.  I'm guessing that's because it's a German machine (judging from the exception) and thus might be named something different.  The second exception is possibly somewhat related where the key is potentially not in English as I had assumed.

I'll dig into this.  I should be able to figure out a more dynamic way to get references to these, but it'll take me a bit.

**Comment by lukasIO** [comment]
If you tell me where to find the user-name in the source and how I'd find what that LocalService User might be called in German, then I'd test with an own build to see if that's the problem!

**Comment by WereDev** [comment]
@lukasIO Ok, I pushed v1.2 which gets the Local Service dynamically from the system which should fix the first issue.  This is in code at https://github.com/WereDev/Wu10Man/blob/642ba33dffc10d5e926148dbf628e588b60a9629/Wu10Man/Editors/ServiceEditor.cs#L56

The Registry issue I'm still scratching my head about a bit.  I'm pretty sure that those keys should exist regardless of language but I still have to do some digging.  The reference for those keys is at https://github.com/WereDev/Wu10Man/blob/642ba33dffc10d5e926148dbf628e588b60a9629/Wu10Man/UserControls/GroupPolicyControl.xaml.cs#L28.

I'm going to keep looking into it though I'm heading out on vacation shortly so it'll be a while before I can get back to it.  Hopefully at least the first fix helps.

**Comment by WereDev** [comment]
@lukasIO I'm an idiot.  I'm not checking to make sure the Registry Key exists.  Some systems haven't populated the AU key yet.  I'm not sure I can get a fix in for it before I head out, but it should be a simple matter of checking if the reg key exists and creating it if it doesn't.  Then I can write values to it.  In the mean time, you can probably manually create the path down to AU and then this should work for you.  I'll get a fix in for it though.

**Comment by WereDev** [comment]
@lukasIO Update v1.2 to also include the Registry Key check which should solve the second issue.

Let me know if this fixes things for you.

**Comment by lukasIO** [comment]
unfortunately not, but the error messages change :)

```
2018-06-24 17:28:27.3956 ERROR In den Registrierungsschlüssel kann nicht geschrieben werden. 
    bei System.ThrowHelper.ThrowUnauthorizedAccessException(ExceptionResource resource)
	   bei Microsoft.Win32.RegistryKey.EnsureWriteable()
	   bei Microsoft.Win32.RegistryKey.SetValue(String name, Object value, RegistryValueKind valueKind)
	   bei WereDev.Utils.Wu10Man.Editors.RegistryEditor.WriteRegistryValue(RegistryKey registryRoot, String registryKey, String registryName, String registryValue, RegistryValueKind registryValueKind) in C:\_Code\Wu10Man\Wu10Man\Editors\RegistryEditor.cs:Zeile 64.
	   bei WereDev.Utils.Wu10Man.Editors.RegistryEditor.WriteLocalMachineRegistryValue(String registryKey, String registryName, String registryValue, RegistryValueKind registryValueKind) in C:\_Code\Wu10Man\Wu10Man\Editors\RegistryEditor.cs:Zeile 26.
	   bei WereDev.Utils.Wu10Man.Editors.ServiceEditor.SetStartupType(ServiceStartMode startMode) in C:\_Code\Wu10Man\Wu10Man\Editors\ServiceEditor.cs:Zeile 20.
	   bei WereDev.Utils.Wu10Man.Editors.ServiceEditor.DisableService() in C:\_Code\Wu10Man\Wu10Man\Editors\ServiceEditor.cs:Zeile 29.
	   bei WereDev.Utils.Wu10Man.UserControls.WindowsServicesControl.DisableService(String serviceName) in C:\_Code\Wu10Man\Wu10Man\UserControls\WindowsServicesControl.xaml.cs:Zeile 57.
	   bei WereDev.Utils.Wu10Man.UserControls.WindowsServicesControl.BtnEnableDisable_Click(Object sender, RoutedEventArgs e) in C:\_Code\Wu10Man\Wu10Man\UserControls\WindowsServicesControl.xaml.cs:Zeile 76.
	   bei System.Windows.RoutedEventHandlerInfo.InvokeHandler(Object target, RoutedEventArgs routedEventArgs)
	   bei System.Windows.EventRoute.InvokeHandlersImpl(Object source, RoutedEventArgs args, Boolean reRaised)
```
It says that I'm not allowed to write in the registry eventhough the dialog pops up to ask for administrative privileges.... strange.

**Comment by WereDev** [comment]
@lukasIO Yeah, registry access is a little bit more wonky for some things than just having Admin access.  I can do some more digging to try to find out why this is different and how to get the info dynamically I did for the user account.  I should be able to get to that this weekend as I think I'll want to spin up a VM for further testing.

**Comment by WereDev** [comment]
@lukasIO Can you tell me which edition of Windows 10 you're using?  I'm looking for if you're running Home, Pro, etc. but also if you're running N or not, which is a European specific build.

Thanks.

**Comment by lukasIO** [comment]
I'm running Win 10 Pro!

**Comment by WereDev** [comment]
@lukasIO Ok, I think I found the issue.  I was being a little foolish with how I was trying to capture exceptions when writing to the registry.  I use this as a mechanism to figure out if I need additional rights or not and since that wasn't being captured correctly, it was unable to get the correct permissions.  I tested this on Win 10 Enterprise and Home.  I didn't have Pro immediately available but it should work there as well.  Fix is in v1.2.1.  Btw, there is also an installer now.
